aptos-move/aptos-aggregator/src/delta_change_set.rs

```diff
@@ -166,6 +166,17 @@ impl DeltaOp {
         Ok(())
     }
 
+    /// Applies next delta on top of self, merging two deltas together. This is a reverse
+    /// of `merge_with_previous_delta`.
+    pub fn merge_with_next_delta(&mut self, next_delta: DeltaOp) -> PartialVMResult<()> {
+        // TODO: Revisit how merge_with_previous_delta is implemented and refactor to
+        // avoid copies/duplication.
+        let mut next_delta = next_delta;
+        next_delta.merge_with_previous_delta(*self)?;
```

Comment:
this involves a copy? not a big deal but just looks ugly, maybe try mem::swap(self, &mut next_delta) and just do the merge_with_previous?